napari/_qt/qt_main_window.py

```diff
@@ -1149,22 +1154,31 @@ def _screenshot(self, flash=True, canvas_only=False) -> 'QImage':
         from .utils import add_flash_animation
 
         if canvas_only:
+            canvas = self._qt_viewer.canvas
+            if size is not None:
+                prev_size = canvas.size
+                canvas.size = size
             img = self._qt_viewer.canvas.native.grabFramebuffer()
             if flash:
                 add_flash_animation(self._qt_viewer._canvas_overlay)
+            if size is not None:
+                canvas.size = prev_size
```

Comment:
maybe put into a try/finally?  if anything goes wrong when grabbing the buffer or running the animation, we've permanently changed the size right?